File: Java/src/main/java/org/codecop/lcdnumbers/Scaling.java
package org.codecop.lcdnumbers;

import java.util.Objects;

/**
 * The scale factor value.
 */
public final class Scaling {

    public static final Scaling NONE = new Scaling(1);
    public static final Scaling TWO = new Scaling(2);

    private final int times;

    public static Scaling of(int t) {
        return new Scaling(t);
    }

    private Scaling(int times) {
        if (times <= 0) {
            throw new IllegalArgumentException("scaling factor must be >= 1");
        }

        this.times = times;
    }

    public boolean none() {
        return times == 1;
    }

    public void times(Runnable block) {
        Objects.requireNonNull(block);

        for (int i = 0; i < times; i++) {
            block.run();
        }
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof Scaling)) {
            return false;
        }
        Scaling that = (Scaling) other;
        return this.times == that.times;
    }

    @Override
    public int hashCode() {
        return times;
    }

    @Override
    public String toString() {
        return Integer.toString(times);
    }
}

File: Kotlin-Spring/src/main/kotlin/org/codecop/lcdnumbers/Scaling.kt
package org.codecop.lcdnumbers

import java.util.Objects

/**
 * The scale factor value.
 */
data class Scaling(private val times: Int) {
    init {
        require(times > 0) { "scaling factor must be >= 1" }
    }
    
    fun none(): Boolean {
        return times == 1
    }
    
    fun times(block: () -> Unit) {
        Objects.requireNonNull(block)
        for (i in 0 until times) {
            block()
        }
    }
    
    override fun toString() = times.toString()
    
    companion object {
        val NONE = Scaling(1)
        val TWO = Scaling(2)
        
        fun of(t: Int): Scaling {
            return Scaling(t)
        }
    }
}